lib/Controller/Report.php

```diff
@@ -0,0 +1,520 @@
+            $data['selectedName'] = $this->mediaFactory->getById($data['selectedId'])->name;
+        }
+
+        $this->getState()->template = 'report-schedule-form-add';
+        $this->getState()->setData($data);
+
+    }
+
+    /**
+     * Report Schedule Edit Form
+     */
+    public function editReportScheduleForm($reportScheduleId)
+    {
+        $reportSchedule = $this->reportScheduleFactory->getById($reportScheduleId);
+
+        $this->getState()->template = 'reportschedule-form-edit';
+        $this->getState()->setData([
+            'reportSchedule' => $reportSchedule
+        ]);
+    }
+
+    /**
+     * Report Schedule Delete Form
+     */
+    public function deleteReportScheduleForm($reportScheduleId)
+    {
+        $reportSchedule = $this->reportScheduleFactory->getById($reportScheduleId);
+
+        if (!$this->getUser()->checkDeleteable($reportSchedule))
+            throw new AccessDeniedException(__('You do not have permissions to delete this report schedule'));
+
+//         var_dump($reportSchedule->name);
+//         die();
+
+        $data = [
+            'reportSchedule' => $reportSchedule
+        ];
+
+        $this->getState()->template = 'reportschedule-form-delete';
+        $this->getState()->setData($data);
+
+    }
+
+    //</editor-fold>
+
+    //<editor-fold desc="Saved report">
+
+    /**
+     * Saved report Grid
+     */
+    public function savedReportGrid()
+    {
+        $reportName = 'summaryReport';
+
+        $savedReports = $this->savedReportFactory->query($this->gridRenderSort(), $this->gridRenderFilter([
+            'saveAs' => $this->getSanitizer()->getString('saveAs'),
+        ]));
+
+        foreach ($savedReports as $savedReport) {
+            /** @var \Xibo\Entity\SavedReport $savedReport */
+
+            $savedReport->includeProperty('buttons');
+
+            $savedReport->buttons[] = [
+                'id' => 'button_show_report.now',
+                'class' => 'XiboRedirectButton',
+                'url' => $this->urlFor('savedreport.preview', ['id' => $savedReport->savedReportId, 'name' => $reportName] ),
+                'text' => __('Preview report')
+            ];
+
+            // Delete
+            $savedReport->buttons[] = [
+                'id' => 'savedReport_delete_button',
+                'url' => $this->urlFor('savedreport.delete.form', ['id' => $savedReport->savedReportId]),
+                'text' => __('Delete')
+            ];
+        }
+        $this->getState()->template = 'grid';
+        $this->getState()->recordsTotal = $this->savedReportFactory->countLast();
+        $this->getState()->setData($savedReports);
+    }
+
+    /**
+     * Displays the Saved Report Page
+     */
+    public function displaySavedReportPage()
+    {
+        // Call to render the template
+        $this->getState()->template = 'saved-report-page';
+        $this->getState()->setData([
+            'savedReports' => $this->savedReportFactory->query()
```

Comment:
it would be better to have this as a separate `grid` type call, so that we can take advantage of paging, etc. If there are 10000 saved reports we wont be able to load them like this.